### PR TITLE
feat: tor browser installation

### PR DESCRIPTION
**Pull request by phoenixhaleofficial**
## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Description

- Added Tor Browser as an option in the Web Browsers tab.

## Testing

- Tested on Arch Linux and Debian.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Review by adamperkowski — CHANGES_REQUESTED** [review]
Please run `cargo xtask docgen` and push the changes.

**Review by adamperkowski — APPROVED** [review]
`torbrowser-launcher` packages are not official. Some users might want to get tor browser directly. See https://support.torproject.org/apt/tor-deb-repo and https://support.torproject.org/rpm.

File: core/tabs/applications-setup/browsers/tor-browser.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install torbrowser-launcher
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --needed --noconfirm torbrowser-launcher
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y torbrowser-launcher
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/browsers/tor-browser.sh:22]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: ${PACKAGER}${RC}"
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/browsers/tor-browser.sh (hunk 1 above)
Comment (phoenixhaleofficial) [reply core/tabs/applications-setup/browsers/tor-browser.sh:22]: @jeevithakannan2 My script is based on the firefox.sh script, where there's also 2 double-quotes.
Same in all other scripts.

Comment (jeevithakannan2) [reply core/tabs/applications-setup/browsers/tor-browser.sh:22]: Yeah working on removing unnecessary quotes. The double quotes is not necessary.

**Comment by phoenixhaleofficial** [comment]
@adamperkowski both instructions relate to only the tor package, not `torbrowser-launcher`

The `torbrowser-launcher` package seems to be maintained by the Tor project, see [https://github.com/torproject/torbrowser-launcher](https://github.com/torproject/torbrowser-launcher)

They also state that the packages in the distro's repos may be out-of-date and recommend installing `torbrowser-launcher` via Flatpak, see [https://github.com/torproject/torbrowser-launcher?tab=readme-ov-file#installing-in-any-linux-distro-using-flatpak](https://github.com/torproject/torbrowser-launcher?tab=readme-ov-file#installing-in-any-linux-distro-using-flatpak)

Maybe it would be best to change the install script to use Flatpak instead of native package managers, what do you think?

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**